samples/msal-node-samples/client-credentials-with-cert-from-key-vault/test/client-credentials-with-cert-from-key-vault-aad.spec.ts

```diff
@@ -0,0 +1,95 @@
+import {
+    RETRY_TIMES,
+    validateCacheLocation,
+    NodeCacheTestUtils,
+} from "e2e-test-utils";
+import { ConfidentialClientApplication } from "@azure/msal-node";
+import config from "../config/AAD.json";
+import { getKeyVaultSecretClient } from "e2e-test-utils/src/KeyVaultUtils";
+import { getCertificateInfo } from "e2e-test-utils/src/CertificateUtils";
+import { LAB_CERT_NAME, LAB_KEY_VAULT_URL } from "e2e-test-utils/src/Constants";
+
+const TEST_CACHE_LOCATION = `${__dirname}/data/aad.cache.json`;
+
+const getClientCredentialsToken = require("../index");
+
+const cachePlugin = require("../../cachePlugin.js")(TEST_CACHE_LOCATION);
+
+const clientCredentialRequestScopes = ["https://graph.microsoft.com/.default"];
+
+describe("Client Credentials AAD Prod Tests", () => {
+    jest.retryTimes(RETRY_TIMES);
+    jest.setTimeout(90000);
+
+    let thumbprint: string;
+    let privateKey: string;
+    let x5c: string;
+    beforeAll(async () => {
+        await validateCacheLocation(TEST_CACHE_LOCATION);
+
+        const keyVaultSecretClient = await getKeyVaultSecretClient(
+            LAB_KEY_VAULT_URL
+        );
+        [thumbprint, privateKey, x5c] = await getCertificateInfo(
+            keyVaultSecretClient,
+            LAB_CERT_NAME
+        );
+
+        config.authOptions.clientId = process.env.AZURE_CLIENT_ID;
+        config.authOptions.authority = `https://login.microsoftonline.com/${process.env.AZURE_TENANT_ID}`;
```

Comment:
TODO: I think I can get these values, and possibly the keyvault name and url from the lab client?

Reply:
For client_credentails, I see that MSAL.NET hardcodes everything - https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Integration.netcore/Infrastructure/ConfidentialAppSettings.cs#L40

It's ok to hardcode client id, tenant id, keyvault names - these are not secrets.